generate/generate.go

```diff
@@ -646,24 +751,83 @@ func hasGenHCLHeader(code string) bool {
 	return false
 }
 
-func checkGeneratedFilesPaths(generated []fileInfo) error {
+func checkGeneratedFilesPaths(rootdir string, stackpath string, generated []fileInfo) error {
 	logger := log.With().
-		Str("action", "checkGeneratedFilesPaths()").
+		Str("action", "generate.checkGeneratedFilesPaths()").
 		Logger()
 
 	logger.Trace().Msg("Checking for invalid paths on generated files.")
 
+	errs := errors.L()
+
 	for _, file := range generated {
-		fname := filepath.ToSlash(file.Name())
-		if strings.Contains(fname, "/") {
-			return errors.E(
-				ErrInvalidFilePath,
-				"filenames with dirs are disallowed, config %q provided filename %q",
-				file.Origin(),
-				file.Name())
+		relFilename := file.Label()
+		if !strings.Contains(relFilename, "/") {
+			continue
+		}
+
+		abspath := filepath.Join(stackpath, relFilename)
+
+		switch {
+		case strings.HasPrefix(relFilename, "/"):
+			errs.Append(errors.E(ErrInvalidGenBlockLabel,
+				"%s: %s: starts with /",
+				file.Origin(), file.Label()))
+			continue
+		case strings.HasPrefix(relFilename, "./"):
```

Comment:
I added this constraint because it makes some things simpler, or else ./name and name are actually the same, so a simple label checking is not enough. Still doable but it seemed OK to constrain on a uniform way to define the paths. But can change this, not a 100% sure it is a good idea.